src/TTCache.php

```diff
@@ -37,39 +39,37 @@ public function __construct(CacheInterface $cache, Closure $keyHasher = null)
         $this->keyHasher = $keyHasher ?? fn ($x) => md5($x);
     }
 
-    protected function hashedKey(string $k) : string
+    protected function hashedKey(string $k): string
     {
-        return 'k-' . ($this->keyHasher)($k);
+        return 'k-'.($this->keyHasher)($k);
     }
 
-    protected function hashedTag(string $t) : string
+    protected function hashedTag(string $t): string
     {
-        return 't-' . ($this->keyHasher)($t);
+        return 't-'.($this->keyHasher)($t);
     }
 
     /**
      * Cache the result of a callback at the given key
      *
-     * @param string $key                       The unique key where this value will be cached
-     * @param callable $cb                      The callback to compute the value to cache
-     * @param int|null $ttl                     How long this value should stay in cache. A ttl applied in a nested
+     * @param  string  $key                       The unique key where this value will be cached
+     * @param  callable  $cb                      The callback to compute the value to cache
+     * @param  int|null  $ttl                     How long this value should stay in cache. A ttl applied in a nested
      *                                          call to `remember` will also apply to any value coming in a wrapping
      *                                          call to the "remember" setting the ttl. If multiple such TTL calls
      *                                          exist in nested calls, the shortest one will win.
-     * @param array<string|TagInterface> $tags A list of tags / surrogate keys that can be used to clear this value
+     * @param  array<string|TagInterface>  $tags A list of tags / surrogate keys that can be used to clear this value
      *                                          out of cache. For example, if many different calls to remember exist
      *                                          in the codebase that render a user's data and all of those are tagged
      *                                          with 'user:1' (1 being the user's id), then using 'user:1' to clear
      *                                          the cache would eliminate all values that were tagged with this from
      *                                          the cache. A nested call to remember that uses tags will have all its
      *                                          tags applied to all wrapping calls to `remember`
      *
-     * @return Result
      * @throws \Throwable
-     * @phpstan-ignore-next-line
      * @throws \Psr\SimpleCache\InvalidArgumentException
      */
-    public function remember(string $key, ?int $ttl, array $tags, callable $cb) : Result
+    public function remember(string $key, callable $cb, array $tags = [], int $ttl = null): Result
```

Comment:
It would be yeah.
This brings things more in line with the vision experience I had for working with this on PHP8+.

But note that upgrading simple-cache to newer versions means dropping the current Redis support as that package is only compatible with version 1 of simple cache.

I would think we would keep maintaining v2 and v3 branches for some time until we can safely rely on working with simplecache 2|3